Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionStorage.kt

```diff
@@ -216,7 +217,17 @@ open class AnalyticsKeySubmissionStorage(
 
     suspend fun clear() {
         Timber.tag(TAG).d("clear()")
-        dataStore.clear()
+        val keys = dataStore.data.first().asMap().keys.filter {
+            if (sharedPrefKeySuffix.isEmpty()) {
+                it.name.endsWith(AnalyticsRAKeySubmissionStorage.sharedPrefKeySuffix).not()
+            } else {
+                it.name.endsWith(AnalyticsRAKeySubmissionStorage.sharedPrefKeySuffix)
+            }
+        }
+        dataStore.edit { prefs -> keys.forEach { key -> prefs.remove(key) } }
+        Timber.d(
+            dataStore.data.first().asMap().keys.toString()
+        )
```

Comment:
clear before was clearing all keys regardless of which Test Type storage triggered `clear` and this lead to what ever test is registered last it will clear pervious test collected data not only the relevant test type